profiles/schema.py

```diff
@@ -0,0 +1,131 @@
+from graphql_jwt.decorators import login_required
+from munigeo.models import AdministrativeDivision
+from thesaurus.models import Concept
+
+from .models import Profile
+
+
+class ConceptType(DjangoObjectType):
+    class Meta:
+        model = Concept
+        fields = ("code",)
+
+    vocabulary = graphene.String()
+    label = graphene.String()
+
+    def resolve_vocabulary(self, info, **kwargs):
+        return self.vocabulary.prefix
+
+
+class AdministrativeDivisionType(DjangoObjectType):
+    class Meta:
+        model = AdministrativeDivision
+        fields = ("children", "origin_id", "ocd_id", "municipality")
+
+    type = graphene.String()
+    name = graphene.String()
+
+    def resolve_children(self, info, **kwargs):
+        return self.children.filter(type__type="sub_district")
+
+    def resolve_type(self, info, **kwargs):
+        return self.type.type
+
+
+class ProfileType(DjangoObjectType):
+    class Meta:
+        model = Profile
+        fields = (
+            "nickname",
+            "image",
+            "email",
+            "phone",
+            "language",
+            "contact_method",
+            "concepts_of_interest",
+            "divisions_of_interest",
+        )
+
+    concepts_of_interest = graphene.List(ConceptType)
+    divisions_of_interest = graphene.List(AdministrativeDivisionType)
+
+    def resolve_concepts_of_interest(self, info, **kwargs):
+        return self.concepts_of_interest.all()
+
+    def resolve_divisions_of_interest(self, info, **kwargs):
+        return self.divisions_of_interest.all()
+
+
+Language = graphene.Enum("Language", [(l[1].upper(), l[0]) for l in settings.LANGUAGES])
+ContactMethod = graphene.Enum(
+    "ContactMethod", [(cm[1].upper(), cm[0]) for cm in settings.CONTACT_METHODS]
+)
+
+
+class ProfileInput(graphene.InputObjectType):
+    nickname = graphene.String()
+    image = graphene.String()
+    email = graphene.String()
+    phone = graphene.String()
+    language = Language()
+    contact_method = ContactMethod()
+    concepts_of_interest = graphene.List(graphene.String)
+    divisions_of_interest = graphene.List(graphene.String)
+
+
+class UpdateProfile(graphene.Mutation):
+    class Arguments:
+        profile = ProfileInput(required=True)
+
+    profile = graphene.Field(ProfileType)
+
+    @login_required
+    def mutate(self, info, **kwargs):
+        profile_data = kwargs.pop("profile")
+        concepts_of_interest = profile_data.pop("concepts_of_interest", [])
+        divisions_of_interest = profile_data.pop("divisions_of_interest", [])
+        profile, created = Profile.objects.get_or_create(user=info.context.user)
+        profile.nickname = profile_data.nickname
+        profile.image = profile_data.image
+        profile.email = profile_data.email
+        profile.phone = profile_data.phone
+        profile.language = profile_data.language
+        profile.contact_method = profile_data.contact_method
+        profile.save()
+
+        cois = []
```

Comment:
you can optimize your queryset to avoid looping altogether:
```python
from django.db.models import CharField, Value as V
from django.db.models.functions import Concat

...

cois = Concept.objects.annotate(
    identifier=Concat(
        "vocabulary__prefix", V(":"), "code", output_field=CharField()
    )
).filter(identifier__in=concepts_of_interest)
profile.concepts_of_interest.set(cois)
```